src/group/managed_group/callbacks.rs

```diff
@@ -22,33 +50,75 @@ pub struct ManagedGroupCallbacks {
     pub(crate) error_occured: Option<ErrorOccured>,
 }
 
-#[allow(clippy::too_many_arguments)]
-impl ManagedGroupCallbacks {
-    pub fn new(
-        validate_add: Option<ValidateAdd>,
-        validate_remove: Option<ValidateRemove>,
-        member_added: Option<MemberAdded>,
-        member_removed: Option<MemberRemoved>,
-        member_updated: Option<MemberUpdated>,
-        app_message_received: Option<AppMessageReceived>,
-        invalid_message_received: Option<InvalidMessageReceived>,
-        error_occured: Option<ErrorOccured>,
-    ) -> Self {
+impl<'a> ManagedGroupCallbacks {
+    pub fn new() -> Self {
         Self {
-            validate_add,
-            validate_remove,
-            member_added,
-            member_removed,
-            member_updated,
-            app_message_received,
-            invalid_message_received,
-            error_occured,
+            validate_add: None,
+            validate_remove: None,
+            member_added: None,
+            member_removed: None,
+            member_updated: None,
+            app_message_received: None,
+            invalid_message_received: None,
+            error_occured: None,
         }
     }
+    /// Validator function for AddProposals
+    pub fn with_validate_add(mut self, validate_add: ValidateAdd) -> Self {
```

Comment:
This is way better already but pretty inflexible. Not too important for now. But for a builder pattern these functions usually have look like this `fn with_validate_add(&mut self, validate_add: ValidateAdd) -> &mut Self ` and then you have `fn build(&self) -> Self` function to build the object. The `build` isn't always necessary (like in this case I think we could skip if we wanted). But not consuming `self` allows different ways of calling the function.

Reply:
Yeah, this is just a constructor pattern obviously. Do you have any pointer to documentation on this, so that I can align on that?

Reply:
These two are pretty good reads for this https://github.com/rust-unofficial/patterns/blob/master/patterns/builder.md https://www.ameyalokare.com/rust/2017/11/02/rust-builder-pattern.html. But I'd happy with keeping it like this for now and improve later if we see that it's not flexible enough.